loopy/kernel/dependency.py

```diff
@@ -0,0 +1,208 @@
+import islpy as isl
+
+from loopy.symbolic import BatchedAccessMapMapper
+from loopy import LoopKernel
+from loopy import InstructionBase
+
+from dataclasses import dataclass
+from enum import Enum
+
+class DependencyType(Enum):
```

Comment:
I'm unsure whether we'll want this, or whether we'll want all those union'd together.